third_party/ibis/ibis_oracle/datatypes.py

```diff
@@ -0,0 +1,155 @@
+# Copyright 2023 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from functools import partial
+from typing import Optional, TypedDict
+
+import sqlalchemy as sa
+import sqlalchemy.types as sat
+from sqlalchemy.dialects import oracle
+from sqlalchemy.dialects.oracle.cx_oracle import OracleDialect_cx_oracle
+
+import ibis.expr.datatypes as dt
+from ibis.backends.base.sql.alchemy.datatypes import ibis_type_to_sqla
+import cx_Oracle
+
+# Update to avoid cast to CLOB/Text
+ibis_type_to_sqla[dt.String] = sa.sql.sqltypes.String(length=4000)
+ibis_type_to_sqla[dt.Float32] = sat.Float(precision=23).with_variant(oracle.FLOAT(), 'oracle')
```

Comment:
The upgrade to SQLalchemy 2.0 required this call to prevent the following exception: `sqlalchemy.exc.ArgumentError: Oracle FLOAT types use 'binary precision', which does not convert cleanly from decimal 'precision'.`

[Here](https://docs.sqlalchemy.org/en/20/dialects/oracle.html#sqlalchemy.dialects.oracle.BINARY_DOUBLE.params.precision) is the alchemy docs link as well.

Reply:
Oracle has it's own dedicated float-32 (BINARY_FLOAT) and float-64 (BINARY_DOUBLE) data types so I think they would be more appropriate to use. But I don't fully understand what `ibis_type_to_sqla` is used for so I'm happy for this conversation to be resolved seeing as integration tests are passing.